### DISCUSSION TITLE
How to react to blur/unfocus in a QCard?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#16126] [General - Components / Directives / etc] [answered]
I have a QCard and would like to trigger an action when it becomes inactive (loses focus)

```html
<q-card bordered flat v-for="pill in loadedPerson?.Pills" :key="pill.ID" class="pill text-center bg-blue-1"
    @blur="updatePill">
    <q-card-section class="q-pb-none">
        <div class="text-h4 text-bold">{{ pill.Name }}</div>
    </q-card-section>
    <q-card-section class="flex row q-pt-none flex-center items-center">
        <q-input maxlength="2" size="2" dense borderless v-model="pill.Amount"
            class="text-h4 text-bold amountPills"></q-input>
        <q-icon size="xl" name="las la-plus-circle" class="cursor-pointer" @click="pill.Amount++"></q-icon>
        <q-icon size="xl" name="las la-minus-circle" class="cursor-pointer" @click="pill.Amount--"></q-icon>
    </q-card-section>
</q-card>
```

I expected `@blur` to trigger `updatePill` when I leave the QCard but no, nothing happens (and no relevant message is emitted either). I also tried `onfocusout="updatePill"` but again - no reaction.

A Codepen is available at https://codepen.io/WoJWoJ/pen/jOQQpyj

**Comment by pdanpdan** [comment]
Make a codepen

**Reply by wsw70** [reply]
Done (added to the question)

**Reply by pdanpdan** [reply, marked as answer]
https://codepen.io/pdanpdan/pen/wvQQxpO?editors=1010

- QCard is not focusable by default (it needts a `tabindex="xxx"`)
- the reactToBlur function is not calling `console.log` but returning it :)
- you had a wrong closing tag for `q-card` (`</qcard>`)

**Comment by wsw70** [comment]
> https://codepen.io/pdanpdan/pen/wvQQxpO?editors=1010

Thank you. Shouldn't `tabindex` be different for each component?

> * QCard is not focusable by default (it needts a `tabindex="xxx"`)

Ah, I did not know `tabindex` at all.


> * the reactToBlur function is not calling `console.log` but returning it :)

This is what happens when you are disturbed by children in the middle of coding :)

> * you had a wrong closing tag for `q-card` (`</qcard>`)

Ah - this is why I was getting such weird results when separating the elements similarly to what you did. Thanks!

**Reply by pdanpdan** [reply]
`tabindex` is a focus priority indicator:

- negative values can be focused by mouse or by code (not kbd)
- 0 or positive are focusable by kbd also
- all elements with same priority are navigated in document order
- priorities are ordered in ascending order (smallest first)
